.github/workflows/nightly.yml

```diff
@@ -31,31 +31,241 @@ jobs:
+          if [ -f app/jakartaee-hello-world/Dockerfile ]; then
+            echo "Dockerfile found, test failed."
+            exit 1
+          fi
+          rm -rf app
+
       - name: Run Archetype for EE 9.1 Web Profile, SE 8
         run: |
           mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=9.1 -Dprofile=web -DjavaVersion=8 -DoutputDirectory=app -Dgoals="clean package"
           rm -rf app
 
+      - name: Run Archetype for EE 9.1 Web Profile, SE 8, with Docker
+        run: |
+          mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=9.1 -Dprofile=web -DjavaVersion=8 -Ddocker=yes -DoutputDirectory=app -Dgoals="clean package"
+          if [ -f app/jakartaee-hello-world/Dockerfile ]; then
+            echo "Dockerfile found, test failed."
+            exit 1
+          fi
+          rm -rf app
+
+      - name: Run Archetype for EE 9.1 Core Profile, SE 8
+        id: ee_9_1_core_se_8_none
+        continue-on-error: true
+        run: |
+          mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=9.1 -Dprofile=core -DjavaVersion=8 -DoutputDirectory=app -Dgoals="clean package"
+
+      - name: Check Maven Build Status
+        run: |
+          if [[ "${{ steps['ee_9_1_core_se_8_none'].outcome }}" != "failure" ]]; then
+            echo "Maven build for EE 9.1 Core Profile, SE 8, without Docker succeeded, test failed."
+            exit 1
+          fi
+
+      - name: Run Archetype for EE 9.1 Core Profile, SE 8, with Docker
+        id: ee_9_1_core_se_8_none_docker
+        continue-on-error: true
+        run: |
+          mvn archetype:generate -DinteractiveMode=false -DaskForDefaultPropertyValues=false -DarchetypeGroupId=org.eclipse.starter -DarchetypeArtifactId=jakarta-starter -DarchetypeVersion=2.3.0-SNAPSHOT -DjakartaVersion=9.1 -Dprofile=core -DjavaVersion=8 -Ddocker=yes -DoutputDirectory=app -Dgoals="clean package"
+
+      - name: Check Maven Build Status
+        run: |
+          if [[ "${{ steps['ee_9_1_core_se_8_none_docker'].outcome }}" != "failure" ]]; then
+            echo "Maven build for EE 9.1 Core Profile, SE 8, with Docker succeeded, test failed."
+            exit 1
+          fi
+
+      - name: Run Archetype for EE 10, SE 8
```

Comment:
Since we know the SE 8 selection will fail based on the Jakarta EE version (10) and not the other parameters, I think we can reduce the number of tests a little. I think lets do this particular check only once for EE 10 version - so probably just keep this one and not the rest of it's kind for EE 10.